storage/blobserviceclient.go

```diff
@@ -69,7 +69,11 @@ func GetContainerReferenceFromSASURI(sasuri url.URL) (*Container, error) {
 	if len(path) <= 1 {
 		return nil, fmt.Errorf("could not find a container in URI: %s", sasuri.String())
 	}
-	cli := newSASClient().GetBlobService()
+	c, err := newSASClientFromURL(&sasuri)
+	if err != nil {
+		return nil, err
+	}
+	cli := c.GetBlobService()
```

Comment:
>GetBlobService [](start = 10, length = 14)

can't SAS be only for specific types of azure storage api's ( i.e. just for queue service ). In that case when is the lack of permission discovered ?

Reply:
should be discovered when the API call is made.

Reply:
hmm it should be discoverable based on the URI I think, so we should be able to fail faster, either way this is the deprecated package so maybe not worth investing here.